src/mscorlib/src/System/String.cs

```diff
@@ -159,27 +159,26 @@ public sealed class String : IComparable, ICloneable, IConvertible, IEnumerable
             Contract.Ensures(Contract.Result<String>() != null);
             Contract.EndContractBlock();
 
-            if (separator == null)
```

Comment:
I assume this was removed because the StringBuilder.Append() correctly handle the null case. Is that correct?

Did this particular change impact the results in a significant way?